provider/blob/src/blob_data_provider.rs

```diff
@@ -137,13 +77,26 @@ impl BufferProvider for BlobDataProvider {
         key: ResourceKey,
         req: &DataRequest,
     ) -> Result<DataResponse<BufferMarker>, DataError> {
-        let yoked_buffer = self.get_file(key, req)?;
         let mut metadata = DataResponseMetadata::default();
         // TODO(#1109): Set metadata.data_langid correctly.
         metadata.buffer_format = Some(BufferFormat::Postcard07);
         Ok(DataResponse {
             metadata,
-            payload: Some(DataPayload::from_yoked_buffer(yoked_buffer)),
+            payload: Some(DataPayload::from_yoked_buffer(
+                self.data
+                    .try_project_cloned_with_capture((key, req), |zm, (key, req), _| {
+                        zm.get(&key.get_hash(), &req.options.write_to_string())
+                            .map_err(|e| {
+                                match e {
+                                    KeyError::K0 => DataErrorKind::MissingResourceKey,
+                                    KeyError::K1 => DataErrorKind::MissingResourceOptions,
+                                }
+                                .with_req(key, req)
+                            })
+                    })?,
```

Comment:
Nit: I prefer the old function `get_file` where the fetching was done in a separate step before attempting to construct the DataPayload, in part because I find it poor style to have hidden `?` operators. I don't feel strongly.